extensions/ql-vscode/src/common/vscode/abstract-webview-view-provider.ts

```diff
@@ -0,0 +1,81 @@
+import * as vscode from "vscode";
+import { Uri, WebviewViewProvider } from "vscode";
+import { WebviewKind, WebviewMessage, getHtmlForWebview } from "./webview-html";
+import { Disposable } from "../disposable-object";
+import { App } from "../app";
+
+export abstract class AbstractWebviewViewProvider<
+  ToMessage extends WebviewMessage,
+  FromMessage extends WebviewMessage,
+> implements WebviewViewProvider
+{
+  protected webviewView: vscode.WebviewView | undefined = undefined;
+  private disposables: Disposable[] = [];
+
+  constructor(
+    private readonly app: App,
+    private readonly webviewKind: WebviewKind,
```

Comment:
Is there a specific reason for choosing to do this differently than the `AbstractWebview`? In there, this gets returned by the `getPanelConfig` method, which also includes other (optional) options. Right now we probably don't need any of those for webview views, but it would be nice to keep it consistent with the `AbstractWebview` and keep the options open (even though it's quite easy to change afterwards).

Reply:
It just seemed like unnecessary complexity for now but I'm open to changing it.

Reply:
I'm fine with keeping it as-is, but we should reconsider if we ever start exposing other options.